internal/app/go-filecoin/connectors/storage_miner/connector.go

```diff
@@ -85,7 +87,24 @@ func (m *StorageMinerNodeConnector) handleNewTipSet(ctx context.Context, previou
 
 // SendSelfDeals creates self-deals and sends them to the network.
 func (m *StorageMinerNodeConnector) SendSelfDeals(ctx context.Context, startEpoch, endEpoch abi.ChainEpoch, pieces ...abi.PieceInfo) (cid.Cid, error) {
-	waddr, err := m.getMinerWorkerAddress(ctx, m.chainState.Head())
+	view, err := m.chainView(ctx, m.chainState.Head())
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	_, waddr, err := view.MinerControlAddresses(ctx, m.minerAddr)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	// ensure miner is in escrow table
+	err = m.establishEscrowBalanceIfNeeded(ctx, view, waddr, m.minerAddr)
```

Comment:
We are (per go-storage-miner) still self-dealing to fill sectors. This requires the miner and miner work have an entry (which can be zero) in the markets escrow table.